main/realmain.cc

```diff
@@ -572,13 +584,18 @@ int realmain(int argc, char *argv[]) {
             }
         }
 
-        gs->errorQueue->flushErrors(true);
+        gs->errorQueue->flushAllErrors(*gs);
 
         if (!opts.noErrorCount) {
-            gs->errorQueue->flushErrorCount();
+            if (gs->errorQueue->nonSilencedErrorCount == 0) {
```

Comment:
Optional: Hmm, since `flushAutocorrects` is still in `ErrorFlusherStdout`, maybe we should move this back too.